torch_frame/data/multi_nested_tensor.py

```diff
@@ -116,41 +116,34 @@ def from_tensor_mat(
 
         return cls(num_rows, num_cols, values, offset)
 
-    def __setitem__(self, index: Any, values: Any):
-        raise RuntimeError(
-            "MultiNestedTensor object does not support setting values. "
-            "It should be used for read-only. ")
-
     def __getitem__(
         self,
         index: Any,
     ) -> Union['MultiNestedTensor', Tensor]:
-
         if isinstance(index, tuple):
             # index[0] for row indexing, index[1] for column indexing
             assert len(index) == 2
             if isinstance(index[0], int) and isinstance(index[1], int):
                 # Returns Tensor
-                return self.get_value(index[0], index[1])
+                return self._get_value(index[0], index[1])
             else:
                 # Returns MultiNestedTensor
-                out = self
                 for dim, idx in enumerate(index):
-                    out = out.select(idx, dim)
-                return out
+                    self = self.select(idx, dim)
```

Comment:
Does this mean that we will modify `self` when doing some indexing?

Reply:
+1

Reply:
It seems that we are not modifying self in-place, since the unit test is passing. @akihironitta could you double-check?